frontend/apps/web/components/site-header/AccountStatusHandler.tsx

```diff
@@ -35,3 +50,65 @@ export function AccountStatusHandler(props: Props) {
     </div>
   );
 }
+
+interface TrialCountdownProps {
+  trialEndDate: number;
+  isAccountStatusValidResp: IsAccountStatusValidResponse | undefined;
+}
+
+function TrialCountdown(props: TrialCountdownProps) {
+  const { trialEndDate, isAccountStatusValidResp } = props;
+
+  const now = Date.now();
+  const daysRemaining = Math.max(
+    0,
+    Math.ceil((trialEndDate - now) / (1000 * 60 * 60 * 24))
+  );
+
+  const isExpired =
+    isAccountStatusValidResp?.accountStatus ==
+    AccountStatus.ACCOUNT_TRIAL_EXPIRED;
+  const isAlmostExpired = daysRemaining <= 3;
```

Comment:
I still think this would be clearer if the date-fns were used here as it's going to handle more edge cases.

```
const daysRemaining = differenceInDays(trailEndDate, now);
const isAlmostExpired = daysRemaining <= 3
```